src/input/gamepad/gamepadCalibration.js

```diff
@@ -0,0 +1,232 @@
+// @flow
+/*eslint indent:0*/
+
+import {Vec2D} from "../../main/util/Vec2D";
+
+import type {ButtonInfo, StickInfo, StickCardinals, TriggerInfo, DPadInfo, GamepadInfo} from "./gamepadInfo";
+import type {Gamepad, Button} from "./gamepad";
+
+function scanForButton ( buttons0 : Array<Button>, buttons1 : Array<Button>
+                       , axes0 : Array<number>, axes1 : Array<number>
+                       ) : null | ButtonInfo {
+  let buttonInfo = null;
+
+  const bLg = buttons1.length;
+
+  for (let i = 0; i < bLg; i++) {
+    if (detectedButtonPressed(buttons0[i].pressed, buttons1[i].pressed)) {
+      buttonInfo = { kind : "pressed", index : i };
+      break;
+    }
+    else if (detectedButtonValue(buttons0[i].value, buttons1[i].value)) {
+      buttonInfo = { kind : "value", index : i, threshold : 0.75 };
+      break;
+    }
+  }
+  if (buttonInfo === null) {
+    const aLg = axes1.length;
+    for (let j = 0; j < aLg; j++) {
+      if (detectedButtonValue(axes0[j], axes1[j])) {
+        buttonInfo = { kind : "axis", index : j, threshold : 0.75 };
+      }
+    }
+  }
+  return buttonInfo;
+};
+
+function detectedButtonPressed ( pressed0 : bool, pressed1 : bool ) : bool {
+  return pressed1 && !pressed0;
+};
+
+function detectedButtonValue ( value0 : number, value1 : number ) : bool {
+  return (value0 < 0.25 && value1 > 0.75);
+};
+
+
+function scanForTrigger ( buttons0 : Array<Button>, buttons1 : Array<Button>
+                        , axes0 : Array<number>, axes1 : Array<number>
+                        ) : null | TriggerInfo {
```

Comment:
Same for triggers: null snapshot at 0, and fully pressed snapshot at 1.

I have not implemented "light trigger" functionality, which is where the user can use a button to serve as a light shield if they do not have any analog triggers. All we need to do is add a new snapshot for a light shield option if the user has a controller without analog triggers.